asdf.sh

```diff
@@ -59,6 +59,60 @@ fi
 _asdf_bin="$ASDF_DIR/bin"
 _asdf_shims="${ASDF_DATA_DIR:-$HOME/.asdf}/shims"
 
+_asdf_should_prepend=no
+if [ -n "${ASDF_FORCE_PREPEND+x}" ]; then
+  _asdf_should_prepend=$ASDF_FORCE_PREPEND
+else
+  # If ASDF_FORCE_PREPEND is not set, then prepend by default on macOS
+  # to workaround `path_helper`.
+  if [ -n "$BASH_VERSION" ] || [ -n "$ZSH_VERSION" ]; then
+    # shellcheck disable=SC3028
```

Comment:
Remind we why we have shellcheck configured with `# shellcheck shell=sh` for this file but then end up disabling a bunch of warnings?

Reply:
I'm assuming this is because this is inside the `[ -n "$BASH_VERSION" ] || [ -n "$ZSH_VERSION" ]`, so we know it's only going to be evaluated by Bash or Zsh. The `else` clause handles all other shells.